server/controller/src/main/java/ai/starwhale/mlops/datastore/parquet/SwReadSupport.java

```diff
@@ -56,6 +58,10 @@ public RecordMaterializer<Map<String, Object>> prepareForRead(Configuration conf
             Map<String, String> metadata,
             MessageType messageType,
             ReadContext readContext) {
+        var errorFlag = metadata.get(ERROR_FLAG_KEY);
+        if (Boolean.parseBoolean(errorFlag)) {
+            throw new SwProcessException(ErrorType.DATASTORE, "the file is invalid, ignore it!");
+        }
```

Comment:
Treat the file with `null` flag as the legacy version?

Reply:
yes